### ISSUE TITLE
Bio.motifs.parse fails on meme files with version 4.10

### ISSUE THREAD

**Issue by mgalardini** [biopython/biopython#461]
just noticed that the current meme version (4.10.0_2) introduced a change in its output file that breaks the biopython parser.

The error is:

```
    >>>from Bio import motifs
    >>>motifs.read(open('meme.txt'), 'meme')
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/usr/local/lib/python2.7/dist-packages/Bio/motifs/__init__.py", line 141, in read
        motifs = parse(handle, format)
      File "/usr/local/lib/python2.7/dist-packages/Bio/motifs/__init__.py", line 77, in parse
        record = meme.read(handle)
      File "/usr/local/lib/python2.7/dist-packages/Bio/motifs/meme.py", line 41, in read
        length, num_occurrences, evalue = __read_motif_statistics(line)
     File "/usr/local/lib/python2.7/dist-packages/Bio/motifs/meme.py", line 219, in __read_motif_statistics
       length = int(ls[3])
    ValueError: invalid literal for int() with base 10: '='
```

Relevant line before:
```
MOTIF  1        width =  19  sites =   3  llr = 43  E-value = 6.9e-002
```
Relevant line now:
```
MOTIF  1 MEME    width =  19  sites =   3  llr = 43  E-value = 6.9e-002
```
A possible ugly workaround would be to change the way this line is parsed (doing splits on 'width' and the other parameters). If you like I can try to implement the fix myself and a couple of tests, just let me know.

p.s. the offending file is here (https://gist.github.com/mgalardini/78cc9d918d3af6cccf22)

**Comment by peterjc** [comment]
Initially reported via the mailing list, http://lists.open-bio.org/pipermail/biopython-dev/2015-January/020817.html

@mdehoon, or @barwil  - do you have a preference for how to update the parser to handle the old and new output?

**Comment by mdehoon** [comment]
I don't have a strong preference. Perhaps the safest is to check for the keywords explicitly.

**Comment by amathelier** [comment]
I suggest to change it with the following working version:

def __read_motif_statistics(line):
    line = line[5:].strip()
    ls = line.split()
    indx_shift = 1 if ls[1] == "MEME" else 0  # meme 4.10 changed the format
    length = int(ls[3 + indx_shift])
    num_occurrences = int(ls[6 + indx_shift])
    evalue = float(ls[12 + indx_shift])
    return length, num_occurrences, evalue

I don't have writing access to the biopython package so cannot do it myself.

**Comment by mdehoon** [comment]
Fixed in commit c6e8d02a6eaafd32480498cda977f61be44c1cbf ; checking for keywords explicitly.
